scripts/install_server_from_src.sh

```diff
@@ -15,67 +14,133 @@ set -e
   exit 2
 fi
 
+header "Removing Any Old GRR Installs and Install GRR From Source"
 if $PYTHON_INSTALL; then
+  header "Installing GRR Dependencies"
+  # Install GRR dependencies
+  $SRC_DIR/scripts/install_script_ubuntu.sh -dy -r $SRC_DIR/requirements.txt
+
   # Remove any old installation and install GRR
   rm -rf /usr/lib/python2.7/dist-packages/grr
+  rm -rf /usr/local/lib/python2.7/dist-packages/grr
   python setup.py build
   python setup.py install
 fi
 
 mkdir -p $INSTALL_PREFIX/usr/sbin
+
+header "Install Configuration Files"
 # Set up default configuration
 mkdir -p $INSTALL_PREFIX/etc/grr
-install -p -m644 config/grr-server.yaml $INSTALL_PREFIX/etc/grr
+
+$INSTALL_CMD $SRC_DIR/config/grr-server.yaml $INSTALL_PREFIX/etc/grr/grr-server.yaml
+
 # Set up upstart scripts
 mkdir -p $INSTALL_PREFIX/etc/default
 mkdir -p $INSTALL_PREFIX/etc/init
-install -p -m644 config/upstart/default/grr-* $INSTALL_PREFIX/etc/default
-install -p -m644 config/upstart/grr-* $INSTALL_PREFIX/etc/init
-# Do not copy the client script across for the server package.
-rm $INSTALL_PREFIX/etc/init/grr-client.conf
 
-# Set up grr-server shared directories and files
-mkdir -p $INSTALL_PREFIX/usr/share/grr/scripts
-mkdir -p $INSTALL_PREFIX/usr/share/grr/binaries
-mkdir -p $INSTALL_PREFIX/usr/share/grr/executables/windows/templates/unzipsfx/
+install -p -m644 $SRC_DIR/config/upstart/default/grr-* $INSTALL_PREFIX/etc/default
 
-# Generate all the template directories.
-for template in darwin linux windows; do
-  mkdir -p $INSTALL_PREFIX/usr/share/grr/executables/${template}/templates/
-  mkdir -p $INSTALL_PREFIX/usr/share/grr/executables/${template}/installers/
-  mkdir -p $INSTALL_PREFIX/usr/share/grr/executables/${template}/config/
+for init_script in $SRC_DIR/config/upstart/grr-*; do
+  init_script_fn=$(basename $init_script)
+  if [ $init_script_fn != 'grr-client.conf' ]; then
+    $INSTALL_CMD ${init_script} $INSTALL_PREFIX/etc/init/${init_script_fn}
+  fi
 done
 
-# Make a copy of the default config so users can refer to it.
-install -p -m644 -T config/grr-server.yaml $INSTALL_PREFIX/usr/share/grr/grr-server.yaml.default
+# Reload Upstart configuration for newly installed services
+initctl reload-configuration
+
+# Generate all the template directories and download client templates.
+for template in darwin linux windows; do
+  mkdir -p $SRC_DIR/executables/${template}/templates/
+  mkdir -p $SRC_DIR/executables/${template}/installers/
+  mkdir -p $SRC_DIR/executables/${template}/config/
+done
 
 if $DOWNLOAD_CLIENT_TEMPLATES; then
-  scripts/download_client_templates.sh
+  header "Downloading Client Templates"
+  $SRC_DIR/scripts/download_client_templates.sh -d $SRC_DIR
 fi
 
-# When run from docker we don't need this section because we've already
-# installed the client templates and put the whole src tree under /usr/share/grr
+# When run from docker we don't need this section because we put the whole src 
+# tree under /usr/share/grr
```

Comment:
yes but in the current implementation the context was sent to docker when the docker script started, and you created the template directories after that in lines above, so they aren't there inside the docker image. When you go to repack they won't be there. I think this should be fixed by reverting the Dockerfile changes.